userconnhandler_tcp.go

```diff
@@ -0,0 +1,91 @@
+package freki
+
+import (
+	"fmt"
+	"net"
+	"runtime/debug"
+
+	"github.com/pkg/errors"
+)
+
+type UserConnServer struct {
+	port      uint
+	processor *Processor
+	log       Logger
+	listener  net.Listener
+}
+
+func NewUserConnServer(port uint) *UserConnServer {
+	return &UserConnServer{
+		port: port,
+	}
+}
+
+func (h *UserConnServer) Port() uint {
+	return h.port
+}
+
+func (h *UserConnServer) Type() string {
+	return "user.tcp"
+}
+
+func (h *UserConnServer) Start(processor *Processor) error {
+	h.processor = processor
+	h.log = h.processor.log
+
+	var err error
+	// TODO: can I be more specific with the bind addr?
+	h.listener, err = net.Listen("tcp", fmt.Sprintf(":%d", h.port))
+
+	if err != nil {
+		return err
+	}
+
+	for {
+		conn, err := h.listener.Accept()
+		if err != nil {
+			h.log.Errorf("[user.tcp] %v", err)
+			continue
+		}
+
+		host, port, _ := net.SplitHostPort(conn.RemoteAddr().String())
+		ck := NewConnKeyByString(host, port)
+		md := h.processor.Connections.GetByFlow(ck)
```

Comment:
Yeah, iirc, in the first build, I didn't do that because the connection key was specific to which direction that packets were going. Will add.